cdap-master-spi/src/main/java/io/cdap/cdap/master/spi/environment/MasterEnvironment.java

```diff
@@ -85,4 +86,11 @@ MasterEnvironmentRunnable createRunnable(MasterEnvironmentRunnableContext contex
    * Returns a {@link Supplier} of {@link TwillRunnerService} for running programs.
    */
   Supplier<TwillRunnerService> getTwillRunnerSupplier();
+
+  /**
+   * Returns a {@link SparkEnvConfig} of
+   */
+  default SparkEnvConfig getSparkConf() {
```

Comment:
we'll need to be able to pass some sort of information into this method, like the set of config files and program and plugin jars. Would be good to define some object to pass in that we can add fields to later.

Reply:
sounds good.

Reply:
> I think it would be better to name this like getSparkSubmitConfig(), since it is specific to args that are needed when submitting a spark program.

sounds good.


> we'll need to be able to pass some sort of information into this method, like the set of config files and program and plugin jars. Would be good to define some object to pass in that we can add fields to later.

files and plugin jars can be set from spark submitter side through `KubeSparkSubmitter`. I will be adding that in next PR. The purpose of this method is to make sure all the spark configs for this environment are set. For example, pod template, master base path etc.